alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs

```diff
@@ -312,15 +331,19 @@ utxoTransition = do
   let Shelley.UTxOState utxo _deposits _fees _ppup = u
 
   let txb = txbody tx
+      inputsAndCollateral =
+        Set.union
+          (getField @"inputs" txb)
+          (getField @"collateral" txb)
 
   inInterval slot (getField @"vldt" txb)
     ?! OutsideValidityIntervalUTxO (getField @"vldt" txb) slot
 
-  not (Set.null (Alonzo.txins @era txb)) ?!# InputSetEmptyUTxO
+  not (Set.null (getField @"inputs" txb)) ?!# InputSetEmptyUTxO
 
   feesOK pp tx utxo -- Generalizes the fee to small from earlier Era's
-  eval (Alonzo.txins @era txb ⊆ dom utxo)
-    ?! BadInputsUTxO (eval (Alonzo.txins @era txb ➖ dom utxo))
+  eval (inputsAndCollateral ⊆ dom utxo)
+    ?! BadInputsUTxO (eval (inputsAndCollateral ➖ dom utxo))
```

Comment:
i missed this check! important : )

Reply:
the change I made to `witsVKeyNeeded` was to include the collateral inputs in the set of keys that require signatures, so that signatures are checked for `inputsAndCollateral` not just `inputs`
I changed the function to have 2 additional things, that ^^ and `reqSignerHashes` to avoid the awkward union in the UTXOW rule of the stuff in the Shelley `witsVKeyNeeded` function and the two additional things